bigquery_etl/view/__init__.py

```diff
@@ -216,6 +222,58 @@ def _valid_view_naming(self):
             return False
         return True
 
+    def target_view_identifier(self, target_project=None):
+        """Return the view identifier after replacing project with target_project.
+
+        Result must be a fully-qualified BigQuery Standard SQL table identifier, which
+        is of the form f"{project_id}.{dataset_id}.{table_id}". dataset_id and table_id
+        may not contain "." or "`". Each component may be a backtick (`) quoted
+        identifier, or the whole thing may be a backtick quoted identifier, but not
+        both. Project IDs must contain 6-63 lowercase letters, digits, or dashes. Some
+        project IDs also include domain name separated by a colon. IDs must start with a
+        letter and may not end with a dash. For more information see also
+        https://github.com/mozilla/bigquery-etl/pull/1427#issuecomment-707376291
+        """
+        if target_project:
+            return self.view_identifier.replace(self.project, target_project, 1)
+        return self.view_identifier
+
+    def has_changes(self, target_project=None):
+        """Determine whether there are any changes that would be published."""
+        client = bigquery.Client()
+        target_view_id = self.target_view_identifier(target_project)
+        try:
+            table = client.get_table(target_view_id)
+        except NotFound:
+            print(f"view {target_view_id} will change: does not exist in BigQuery")
+            return True
+
+        expected_view_query = CREATE_VIEW_PATTERN.sub(
+            "", sqlparse.format(self.content, strip_comments=True), count=1
+        ).strip()
+        actual_view_query = sqlparse.format(
+            table.view_query, strip_comments=True
+        ).strip()
+        if expected_view_query != actual_view_query:
+            print(f"view {target_view_id} will change: query does not match")
+            return True
+
+        # check schema
+        schema_file = Path(self.path).parent / "schema.yaml"
```

Comment:
Unless `schema.yaml` is added manually or at some point `bqetl query schema update` is called, most views won't have a schema file.

Reply:
Simply deploying them might be faster though than dry running the view in order to get the schema information

Reply:
yeah, but if it's there we apply it during the publish step